test/e2e/kube-ovn/ipam/ipam.go

```diff
@@ -509,4 +509,56 @@ var _ = framework.Describe("[group:ipam]", func() {
 		deploy = deployClient.PatchSync(deploy, patchedDeploy)
 		checkFn()
 	})
+
+	framework.ConformanceIt("should allocate right IPs for the statefulset when there are multiple IP Pools added to its namespace", func() {
+		replicas := 1
+		ipsCount := 12
+		ipsRange1 := framework.RandomIPPool(cidr, ipsCount)
+		ipsRange2 := framework.RandomIPPool(cidr, ipsCount)
+		testStsName := "test-statefulset"
+		testSubnetName := "ip-pool-subnet2"
+		testIPPool1Name := "ip-pool1"
+		testIPPool2Name := "ip-pool2"
+
+		ginkgo.By("Creating a new subnet " + testSubnetName)
+		testCidr := framework.RandomCIDR(f.ClusterIPFamily)
+		testSubnet := framework.MakeSubnet(testSubnetName, "", testCidr, "", "", "", nil, nil, []string{namespaceName})
+		subnetClient.CreateSync(testSubnet)
+
+		ginkgo.By("Creating IPPool resources ")
+		ippool1 := framework.MakeIPPool(testIPPool1Name, subnetName, ipsRange1, []string{namespaceName})
+		ippool2 := framework.MakeIPPool(testIPPool2Name, testSubnetName, ipsRange2, []string{namespaceName})
+		ippoolClient.CreateSync(ippool1)
+		ippoolClient.CreateSync(ippool2)
+
+		ginkgo.By("Creating statefulset " + testStsName + " with logical switch annotation and no ippool annotation")
+		labels := map[string]string{"app": testStsName}
+		sts := framework.MakeStatefulSet(testStsName, testStsName, int32(replicas), labels, framework.PauseImage)
+		sts.Spec.Template.Annotations = map[string]string{util.LogicalSwitchAnnotation: subnetName}
+		sts = stsClient.CreateSync(sts)
+
+		ginkgo.By("Getting pods for statefulset " + testStsName)
+		pods := stsClient.GetPods(sts)
+		framework.ExpectHaveLen(pods.Items, replicas)
+
+		for _, pod := range pods.Items {
+			framework.ExpectHaveKeyWithValue(pod.Annotations, util.AllocatedAnnotation, "true")
+			framework.ExpectHaveKeyWithValue(pod.Annotations, util.CidrAnnotation, subnet.Spec.CIDRBlock)
+			framework.ExpectHaveKeyWithValue(pod.Annotations, util.GatewayAnnotation, subnet.Spec.Gateway)
+			framework.ExpectHaveKeyWithValue(pod.Annotations, util.LogicalSwitchAnnotation, subnetName)
+			framework.ExpectIPInCIDR(pod.Annotations[util.IPAddressAnnotation], subnet.Spec.CIDRBlock)
+			framework.ExpectMAC(pod.Annotations[util.MacAddressAnnotation])
+			framework.ExpectHaveKeyWithValue(pod.Annotations, util.RoutedAnnotation, "true")
+		}
+
+		ginkgo.By("Deleting statefulset " + testStsName)
+		stsClient.DeleteSync(testStsName)
+
+		ginkgo.By("Deleting ippools")
+		ippoolClient.DeleteSync(testIPPool1Name)
+		ippoolClient.DeleteSync(testIPPool2Name)
+
+		ginkgo.By("Deleting subnet " + testSubnetName)
+		subnetClient.DeleteSync(testSubnetName)
```

Comment:
```suggestion
```

Since we are using `ginkgo.DeferCleanup()`, there is no need to delete the resources here.